hydra-node/src/Hydra/Network/Heartbeat.hs

```diff
@@ -28,21 +28,21 @@ import Hydra.Network.Message (Message (Connected, Disconnected))
 data HeartbeatState = HeartbeatState
   { -- | The map of known 'Connected' parties with the last time they've been "seen".
     -- This is updated when we see a message from another node
-    alive :: Map NodeId Time
+    alive :: !(Map NodeId Time)
   , -- | The set of known parties which might be 'Disconnected'
     -- This is updated after some time no message has been received from a node.
-    suspected :: Set NodeId
+    suspected :: !(Set NodeId)
   , -- | The timestamp of the last sent message.
-    lastSent :: Maybe Time
+    lastSent :: !(Maybe Time)
```

Comment:
This is pointless strictness. The `Maybe` may still contain thunks. The same goes for non-strict `Set` and `Map` above.